src/main/java/org/assertj/core/error/BasicErrorMessageFactory.java

```diff
@@ -161,4 +162,31 @@ public String toString() {
                   CONFIGURATION_PROVIDER.representation().toStringOf(arguments));
   }
 
+  /**
+   * Return the specific name that will be used in the error message.
+   *
+   * @param actual the object that is being verified .
+   * @return an String array of length two. The first string is the object (class) name, the second string is the element name.
+   */
+  public static String[] getSpecificName(Object actual) {
```

Comment:
let's return a custom type as `String[]` does not express what it contains, I suggest `GroupTypeDescription` with two fields `groupTypeName` and `elementTypeName`.
and rename to `getGroupTypeDescription` (or whatever class we'll agree on to return)

Reply:
I also wonder if we could put that code in an interface default method and let `ShouldContain` and co implement it instead of the common base class.
interface could be `WithGroupTypeDescription` (happy to get a better name)

Reply:
Yes, I think an interface will be better.

Reply:
I think that the method ``getGroupTypeDescription`` should be `` static`` rather than ``default`` because non-static method cannot be invoked in static method and ShouldContain is a Class with only static methods.
In this case, an interface may not be needed.  I think we can  make this a static method in ``GroupTypeDescription ``  class  and if other ShouldChontainXXX classes need different implementation, they write their own ``getGroupTypeDescription`` method.

Reply:
Fine by me!